rask-engine/src/resources/mod.rs

```diff
@@ -0,0 +1,38 @@
+/*!
+The resource management system for the ratatosk game engine.
+
+# Example
+
+```
+use lazy_static::lazy_static;
+# use rask_engine::resources::*;
+
+lazy_static! {
+    static ref LIB: ResourceTable = unsafe { ResourceTable::new(0, 0) };
+}
+
+fn test() {
+    unsafe {
+        let _texture: &Texture = LIB.get(0).unwrap();
+    }
+}
+```
+*/
+
+mod library;
+pub mod sound;
+pub mod texture;
+
+#[doc(inline)]
+pub use library::*;
+#[doc(inline)]
+pub use sound::Sound;
+#[doc(inline)]
+pub use texture::Texture;
+
+pub enum Resource {
+    None,
+    Texture(Texture),
+    Sound(Sound),
+    Skeleton(spine::skeleton::Skeleton),
```

Comment:
Wouldn't it be better to box the skeleton?

Reply:
the skeleton is a box in of itself and not larger than texture.
A box would just introduce complexity and overhead

Reply:
Right, it is internally just a pointer